apps-client/mobile/components/election/election-card.tsx

```diff
@@ -440,3 +472,61 @@ function ElectionNotification(props: ElectionNotificationProps) {
   )
   return props.children({ isEnabled, setIsEnabled })
 }
+
+function ElectionRegisterButton(props: ElectionCardProps) {
+  const electionRegisterMutation = reactQueryClient.useMutation(
+    'post',
+    '/elections/:electionId/register',
+    {}
+  )
+  const [isRegistered, setIsRegistered] = useState(props.election.isRegistered)
+  return (
+    <Button
+      size="sm"
+      className="flex-1"
+      disabled={
+        isRegistered ||
+        electionRegisterMutation.isPending ||
+        dayjs().isBefore(props.election.openRegister)
+      }
+      onPress={() => {
+        if (isRegistered) return
+        setIsRegistered(true)
+        electionRegisterMutation.mutateAsync(
+          {
+            pathParams: { electionId: props.election.id },
+            body: { type: 'ONLINE' },
+          },
+          {
+            onError: (error) => {
+              console.error('Election register error', JSON.stringify(error))
+              setIsRegistered(false)
+            },
+          }
+        )
+      }}
+    >
+      <Icon icon={UserRoundCheckIcon} size={16} />
+      <Text>{isRegistered ? 'ลงทะเบียนแล้ว' : 'ลงทะเบียนเลือกตั้งออนไลน์'}</Text>
+    </Button>
+  )
+}
+
+function ElectionRegisterWarning(props: ElectionCardProps) {
+  if (
+    props.election.type !== 'HYBRID' ||
+    props.election.status !== 'NOT_OPENED_VOTE' ||
+    dayjs().isBefore(props.election.closeRegister) ||
+    props.election.isRegistered
+  ) {
+    return null
+  }
```

Comment:
Should warning condition be these instead ?

```suggestion
  if (
    props.election.type !== 'HYBRID' ||
	dayjs().isAfter(props.election.closeRegister) ||
    dayjs().isBefore(props.election.openRegister) ||
    props.election.isRegistered
  ) {
    return null
  }
```

Reply:
sure